src/Features/CSharp/Portable/Completion/CSharpCompletionService.cs

```diff
@@ -65,4 +67,10 @@ internal override CompletionRules GetRules(CompletionOptions options)
 
         return newRules;
     }
+
+    internal override async Task<bool> IsSpeculativeTypeParameterContextAsync(Document document, int position, CancellationToken cancellationToken)
+    {
+        var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+        return CompletionUtilities.IsSpeculativeTypeParameterContext(syntaxTree, position, semanticModel: null, inMemberDeclarationOnly: true, cancellationToken);
```

Comment:
nit: getting a semantic model is cheap.  just get the semantic model here, and pass that into the helper.  the helper doesn't need to take the syntaxTree then.

Reply:
Discussed offline, keeping it as is because we'd need a frozen partial doc otherwise